app/src/test/java/echoserver/client/EchoClientTest.java

```diff
@@ -0,0 +1,33 @@
+package echoserver.client;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import org.junit.Test;
+
+public class EchoClientTest {
+  ClientSocketInterface mockClientSocketInterface = mock(ClientSocketInterface.class);
+  EchoClient testClient = new EchoClient(mockClientSocketInterface);
+  Socket mockClientSocket = mock(Socket.class);
+  InetAddress host = InetAddress.getByName("localhost");
+  int port = 1234;
+
+  public EchoClientTest() throws UnknownHostException {
+  }
+
+  @Test
+  public void connectsToServerAtCorrectAddress() throws IOException {
+    testClient.start(host, port);
+    verify(mockClientSocketInterface).connect(host, port);
+  }
+
+  @Test
+  public void closesConnection() throws IOException {
+    //testClient.start(host, port);
+    //verify(mockClientSocketInterface).close(mockClientSocket);
+  }
```

Comment:
To make this test work, you need to mock the returned value from the `connect` method. 

```suggestion
  public void closesConnection() throws IOException {
    when(mockClientSocketInterface.connect(host, port)).thenReturn(mockClientSocket);
    testClient.start(host, port);
    verify(mockClientSocketInterface).close(mockClientSocket);
  }
```